### PR TITLE
RDM-10838 - SpringBootTest app context caching for reduced integration testing time

### PR DESCRIPTION
**Pull request by sullivangareth**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-10838

### Change description ###

Reduce the separate Spring Boot contexts created for integration tests.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/ccd/BaseTest.java [hunk 1]
```diff
@@ -126,9 +126,13 @@
     @Mock
     protected SecurityContext securityContext;
 
+    @Inject
+    private CacheManager cacheManager;
+
     @Before
     @BeforeEach
     public void initMock() throws IOException {
+        cacheManager.getCacheNames().stream().forEach(cacheName -> cacheManager.getCache(cacheName).clear());
```
Comment (smathangi) [inline src/test/java/uk/gov/hmcts/ccd/BaseTest.java:135]: Can't we disable cache using spring boot properties in tests?

**Comment by smathangi** [comment]
Nice one @garethheath  much waited PR 👍

File: src/test/java/uk/gov/hmcts/ccd/BaseTest.java (hunk 1 above)
Comment (sullivangareth) [reply src/test/java/uk/gov/hmcts/ccd/BaseTest.java:135]: Good point, I had got so intent on getting everything to work as is with minimal changes - but yes that makes perfect sense

File: src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseDetailsEndpointUserRolesIT.java [hunk 2]
```diff
@@ -178,6 +180,7 @@ public void shouldReturn404WhenPostCreateCaseWithNoCreateCaseAccessOnCreatorRole
             CASE_TYPE_CREATOR_ROLE_NO_CREATE_ACCESS);
     }
 
+    @DirtiesContext(methodMode = BEFORE_METHOD)
```
Comment (smathangi) [inline src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseDetailsEndpointUserRolesIT.java:183]: any specific reason

File: Jenkinsfile_CNP [hunk 3]
```diff
@@ -116,7 +116,7 @@ withPipeline(type, product, component) {
             environmentOfDependencies = "aat"
         }
         env.DEFINITION_STORE_HOST      = "http://ccd-definition-store-api-${environmentOfDependencies}.service.core-compute-${environmentOfDependencies}.internal"
-        env.ELASTIC_SEARCH_FTA_ENABLED = "true"
+        env.ELASTIC_SEARCH_FTA_ENABLED = "false"
```
Comment (smathangi) [inline Jenkinsfile_CNP:119]: i guess you're going to revert these ES changes before merge

**Review by smathangi — APPROVED** [review]
LGTM. 🎖️

File: src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseDetailsEndpointUserRolesIT.java (hunk 2 above)
Comment (sullivangareth) [reply src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseDetailsEndpointUserRolesIT.java:183]: I just couldn't get it to work !

File: Jenkinsfile_CNP (hunk 3 above)
Comment (sullivangareth) [reply Jenkinsfile_CNP:119]: Yes - this was just to get a green build on pipeline - will revert before merge

**Review by danlysiak — APPROVED** [review]
Looks like some really nice improvements in testing time 😄 👍

**Review by sumansreenivasan — APPROVED** [review]
Looks good for me

**Comment by MSancaktutar** [comment]
Merged as requested by QA.

**Pull request merged**